irohad/ordering/impl/on_demand_ordering_gate.cpp

```diff
@@ -48,17 +38,48 @@ OnDemandOrderingGate::OnDemandOrderingGate(
       transaction_limit_(transaction_limit),
       ordering_service_(std::move(ordering_service)),
       network_client_(std::move(network_client)),
-      events_subscription_(events.subscribe([this](auto event) {
-        consensus::Round current_round =
-            visit_in_place(event, [this, &current_round](const auto &event) {
-              log_->debug("{}", event);
-              return event.round;
-            });
+      block_events_subscription_(block_events.subscribe([this](auto block) {
+        // block committed, remove transactions from cache
```

Comment:
Can both subscriptions invoke from different threads? If yes, there will be required a synchronization.

Reply:
cache is thread safe. is there anything else requiring sync?